src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -125,61 +129,49 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
         return checksums.Checksum;
     }
 
-    /// <param name="projectsToInclude">Cone of projects to compute a checksum for.  Pass in <see langword="null"/>
-    /// to get a checksum for the entire solution</param>
+    /// <param name="projectConeId">Cone of projects to compute a checksum for.  Pass in <see langword="null"/> to get a
+    /// checksum for the entire solution</param>
     private async Task<SolutionStateChecksums> ComputeChecksumsAsync(
-        HashSet<ProjectId>? projectsToInclude,
+        ProjectId? projectConeId,
         CancellationToken cancellationToken)
     {
+        using var projectCone = SharedPools.Default<HashSet<ProjectId>>().GetPooledObject();
+        AddProjectCone(projectConeId, projectCone.Object);
+
         try
         {
             using (Logger.LogBlock(FunctionId.SolutionState_ComputeChecksumsAsync, this.FilePath, cancellationToken))
             {
                 // get states by id order to have deterministic checksum.  Limit expensive computation to the
                 // requested set of projects if applicable.
                 var orderedProjectIds = GetOrCreateSortedProjectIds(this.ProjectIds);
-                var projectChecksumTasks = orderedProjectIds
-                    .Select(id => (state: this.ProjectStates[id], mustCompute: projectsToInclude == null || projectsToInclude.Contains(id)))
-                    .Where(t => RemoteSupportedLanguages.IsSupported(t.state.Language))
-                    .Select(async t =>
-                    {
-                        // if it's a project that's specifically in the sync'ed cone, include this checksum so that
-                        // this project definitely syncs over.
-                        if (t.mustCompute)
-                            return await t.state.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-
-                        // If it's a project that is not in the cone, still try to get the latest checksum for it if
-                        // we have it.  That way we don't send over a checksum *without* that project, causing the
-                        // OOP side to throw that project away (along with all the compilation info stored with it).
-                        if (t.state.TryGetStateChecksums(out var stateChecksums))
-                            return stateChecksums;
```

Comment:
the chunk from 151 to 155 went away.  we're no longer doing this for cone syncs.  instead, we know we're a cone sync and we only include the project cone checksums we care about.  on the remote side, we also use that info to only mutate that subportion of the solution

Reply:
becausde the OOP side won't drop projects not mentioned in the cone-sync-call, we don't have to worry about this.